sql/catalog/externalcatalog: TestExtractIngestExternalCatalog failed

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://mesolite.cluster.engflow.com/invocations/default/50a7e0a2-b423-44d0-b31f-aaeaa6687356?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvZXh0ZXJuYWxjYXRhbG9nOmV4dGVybmFsY2F0YWxvZ190ZXN0) on master @ [f18cf53e67c1c6eab453eb7be40743b86cdbf349](https://github.com/cockroachdb/cockroach/commits/f18cf53e67c1c6eab453eb7be40743b86cdbf349):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestExtractIngestExternalCatalog1496863546
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    external_catalog_test.go:174: -- test log scope end --
--- FAIL: TestExtractIngestExternalCatalog (6.98s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0xc008fa9200 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (0.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47227

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744316?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744316?buildTab=artifacts#/) on master @ [563e5d37497a78e2781991797f534bdf57b27aa1](https://github.com/cockroachdb/cockroach/commits/563e5d37497a78e2781991797f534bdf57b27aa1):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog3242915175
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    external_catalog_test.go:174: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog3242915175
--- FAIL: TestExtractIngestExternalCatalog (1.85s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0x40074839e0 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (0.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744480?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744480?buildTab=artifacts#/) on master @ [bce8eb6edbbe879d376ab0c99e9d1129e5afef96](https://github.com/cockroachdb/cockroach/commits/bce8eb6edbbe879d376ab0c99e9d1129e5afef96):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog281292332
    test_log_scope.go:76: use -show-logs to present logs inline
    external_catalog_test.go:174: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog281292332
--- FAIL: TestExtractIngestExternalCatalog (1.57s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0x4006819e60 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (0.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744604?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744604?buildTab=artifacts#/) on master @ [bce8eb6edbbe879d376ab0c99e9d1129e5afef96](https://github.com/cockroachdb/cockroach/commits/bce8eb6edbbe879d376ab0c99e9d1129e5afef96):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog1965401694
    test_log_scope.go:76: use -show-logs to present logs inline
    external_catalog_test.go:174: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog1965401694
--- FAIL: TestExtractIngestExternalCatalog (1.94s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0x4007365e60 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (0.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744620?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18744620?buildTab=artifacts#/) on master @ [bce8eb6edbbe879d376ab0c99e9d1129e5afef96](https://github.com/cockroachdb/cockroach/commits/bce8eb6edbbe879d376ab0c99e9d1129e5afef96):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog2360935590
    test_log_scope.go:76: use -show-logs to present logs inline
    external_catalog_test.go:174: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7ae1ff9151ff3b6ba185ef201a529d16/logTestExtractIngestExternalCatalog2360935590
--- FAIL: TestExtractIngestExternalCatalog (1.40s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0x40040bc000 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (0.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://tanzanite.cluster.engflow.com/invocations/default/296dc362-7481-4b36-8548-a4c9fef8278b?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvZXh0ZXJuYWxjYXRhbG9nOmV4dGVybmFsY2F0YWxvZ190ZXN0) on master @ [871fb9ff40888d77373b2da06c991e7f13d82789](https://github.com/cockroachdb/cockroach/commits/871fb9ff40888d77373b2da06c991e7f13d82789):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestExtractIngestExternalCatalog1266586337
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    external_catalog_test.go:174: -- test log scope end --
test logs left over in: outputs.zip/logTestExtractIngestExternalCatalog1266586337
--- FAIL: TestExtractIngestExternalCatalog (26.64s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0xc00e431e60 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (2.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/externalcatalog.TestExtractIngestExternalCatalog [failed](https://tanzanite.cluster.engflow.com/invocations/default/aae51751-59c0-41e3-ba8f-ec7eabdcf020?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvZXh0ZXJuYWxjYXRhbG9nOmV4dGVybmFsY2F0YWxvZ190ZXN0) on master @ [bce8eb6edbbe879d376ab0c99e9d1129e5afef96](https://github.com/cockroachdb/cockroach/commits/bce8eb6edbbe879d376ab0c99e9d1129e5afef96):


```
=== RUN   TestExtractIngestExternalCatalog
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestExtractIngestExternalCatalog1144430647
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    external_catalog_test.go:174: -- test log scope end --
test logs left over in: outputs.zip/logTestExtractIngestExternalCatalog1144430647
--- FAIL: TestExtractIngestExternalCatalog (4.81s)
=== RUN   TestExtractIngestExternalCatalog/udt
    external_catalog_test.go:154: error scanning '&{0xc0026199e0 <nil>}': pq: referenced descriptor ID 109: looking up ID 109: descriptor not found
    --- FAIL: TestExtractIngestExternalCatalog/udt (0.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExtractIngestExternalCatalog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/externalcatalog/external_catalog.go
+++ b/pkg/sql/catalog/externalcatalog/external_catalog.go
@@ -256,6 +256,9 @@ func DropIngestedExternalCatalog(
 		if err := descsCol.WriteDescToBatch(ctx, kvTrace, t, b); err != nil {
 			return errors.Wrap(err, "writing dropping table to batch")
 		}
+		if err := descsCol.DeleteNamespaceEntryToBatch(ctx, kvTrace, t, b); err != nil {
+			return errors.Wrap(err, "writing namespace delete to batch")
+		}
 	}
 	return txn.KV().Run(ctx, b)
 }

